arch/xtensa/src/esp32s2/esp32s2_psram.c

```diff
@@ -0,0 +1,725 @@
+  WRITE_PERI_REG(SPI_MEM_USER2_REG(spi_num), backup_usr2);
+  WRITE_PERI_REG(SPI_MEM_CTRL_REG(spi_num), backup_ctrl);
+}
+
+/* exit QPI mode(set back to SPI mode) */
+
+static void psram_disable_qio_mode(int spi_num)
+{
+  psram_exec_cmd(spi_num, PSRAM_CMD_QPI,
+                 PSRAM_EXIT_QMODE, 8,   /* command and command bit len */
+                 0, 0,                  /* address and address bit len */
+                 0,                     /* dummy bit len */
+                 NULL, 0,               /* tx data and tx bit len */
+                 NULL, 0,               /* rx data and rx bit len */
+                 CS_PSRAM_SEL,          /* cs bit mask */
+                 false);                /* if is program/erase operation */
+}
+
+/* switch psram burst length(32 bytes or 1024 bytes)
+ * datasheet says it should be 1024 bytes by default
+ */
+
+static void psram_set_wrap_burst_length(int spi_num, psram_cmd_mode_t mode)
+{
+  psram_exec_cmd(spi_num, mode,
+                 PSRAM_SET_BURST_LEN, 8, /* command and command bit len */
+                 0, 0,                   /* address and address bit len */
+                 0,                      /* dummy bit len */
+                 NULL, 0,                /* tx data and tx bit len */
+                 NULL, 0,                /* rx data and rx bit len */
+                 CS_PSRAM_SEL,           /* cs bit mask */
+                 false);                 /* if is program/erase operation */
+}
+
+/* send reset command to psram, in spi mode */
+
+static void psram_reset_mode(int spi_num)
+{
+  psram_exec_cmd(spi_num, PSRAM_CMD_SPI,
+  PSRAM_RESET_EN, 8,                /* command and command bit len */
+  0, 0,                             /* address and address bit len */
+  0,                                /* dummy bit len */
+  NULL, 0,                          /* tx data and tx bit len */
+  NULL, 0,                          /* rx data and rx bit len */
+  CS_PSRAM_SEL,                     /* cs bit mask */
+  false);                           /* whether is program/erase operation */
+
+  psram_exec_cmd(spi_num, PSRAM_CMD_SPI,
+  PSRAM_RESET, 8,                   /* command and command bit len */
+  0, 0,                             /* address and address bit len */
+  0,                                /* dummy bit len */
+  NULL, 0,                          /* tx data and tx bit len */
+  NULL, 0,                          /* rx data and rx bit len */
+  CS_PSRAM_SEL,                     /* cs bit mask */
+  false);                           /* whether is program/erase operation */
+}
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+int psram_enable_wrap(uint32_t wrap_size)
+{
+  static uint32_t current_wrap_size;
+
+  if (current_wrap_size == wrap_size)
+    {
+      return OK;
+    }
+
+  switch (wrap_size)
+    {
+      case 32:
+      case 0:
```

Comment:
```suggestion
      case 0:
      case 32:
```
minor style just to have the same as in `psram_support_wrap_size`